src/app/components/elements/Voting.jsx

```diff
@@ -117,7 +117,9 @@ class Voting extends React.Component {
                         (is_comment ? '-comment' : '')
                 );
                 const castToNegative = up ? 1 : -1;
-                weight = Number(saved_weight) * castToNegative;
+                if (saved_weight !== null) {
```

Comment:
Why is saved weight being read on submit? Shouldn't it be read on mount/activation then saved on submit?

Reply:
not sure what difference that would make?

Reply:
I don't understand why it's being read here in addition to `syncSliderWeight` (which gets called on mount and dropdown onshow). The logic looks redundant and wouldn't have it already been read by this point?